codersdk/flags.go

```diff
@@ -0,0 +1,136 @@
+	Oauth2GithubAllowSignups         BoolFlag        `json:"oauth2_github_allow_signups"`
+	Oauth2GithubEnterpriseBaseURL    StringFlag      `json:"oauth2_github_enterprise_base_url"`
+	OidcAllowSignups                 BoolFlag        `json:"oidc_allow_signups"`
+	OidcClientID                     StringFlag      `json:"oidc_client_id"`
+	OidcClientSecret                 StringFlag      `json:"oidc_cliet_secret"`
+	OidcEmailDomain                  StringFlag      `json:"oidc_email_domain"`
+	OidcIssuerURL                    StringFlag      `json:"oidc_issuer_url"`
+	OidcScopes                       StringArrayFlag `json:"oidc_scopes"`
+	TailscaleEnable                  BoolFlag        `json:"tailscale_enable"`
+	TelemetryEnable                  BoolFlag        `json:"telemetry_enable"`
+	TelemetryTraceEnable             BoolFlag        `json:"telemetry_trace_enable"`
+	TelemetryURL                     StringFlag      `json:"telemetry_url"`
+	TLSEnable                        BoolFlag        `json:"tls_enable"`
+	TLSCertFiles                     StringArrayFlag `json:"tls_cert_files"`
+	TLSClientCAFile                  StringFlag      `json:"tls_client_ca_file"`
+	TLSClientAuth                    StringFlag      `json:"tls_client_auth"`
+	TLSKeyFiles                      StringArrayFlag `json:"tls_key_tiles"`
+	TLSMinVersion                    StringFlag      `json:"tls_min_version"`
+	TraceEnable                      BoolFlag        `json:"trace_enable"`
+	SecureAuthCookie                 BoolFlag        `json:"secure_auth_cookie"`
+	SSHKeygenAlgorithm               StringFlag      `json:"ssh_keygen_algorithm"`
+	AutoImportTemplates              StringArrayFlag `json:"auto_import_templates"`
+	MetricsCacheRefreshInterval      DurationFlag    `json:"metrics_cache_refresh_interval"`
+	AgentStatRefreshInterval         DurationFlag    `json:"agent_stat_refresh_interval"`
+	Verbose                          BoolFlag        `json:"verbose"`
+	AuditLogging                     BoolFlag        `json:"audit_logging"`
+	BrowserOnly                      BoolFlag        `json:"browser_only"`
+	ScimAuthHeader                   StringFlag      `json:"scim_auth_header"`
+	UserWorkspaceQuota               IntFlag         `json:"user_workspace_quota"`
+}
+
+type StringFlag struct {
+	Name        string `json:"name"`
+	Flag        string `json:"flag"`
+	EnvVar      string `json:"env_var"`
+	Shorthand   string `json:"shorthand"`
+	Description string `json:"description"`
+	Enterprise  bool   `json:"enterprise"`
+	Default     string `json:"default"`
+	Value       string `json:"value"`
+}
+
+type BoolFlag struct {
+	Name        string `json:"name"`
+	Flag        string `json:"flag"`
+	EnvVar      string `json:"env_var"`
+	Shorthand   string `json:"shorthand"`
+	Description string `json:"description"`
+	Enterprise  bool   `json:"enterprise"`
+	Default     bool   `json:"default"`
+	Value       bool   `json:"value"`
+}
```

Comment:
I might embed a `Flag` struct here instead to reduce some duplication... eg

```
type Flag struct {
	Name        string `json:"name"`
	Flag        string `json:"flag"`
	EnvVar      string `json:"env_var"`
	Shorthand   string `json:"shorthand"`
	Description string `json:"description"`
	Enterprise  bool   `json:"enterprise"`
}

type BoolFlag struct {
        Flag
	Default     bool   `json:"default"`
	Value       bool   `json:"value"`
}

Reply:
Yeah I intentional didn't embed here, I really don't like embedding as a personal preference unless I see a good reason, this seemed simple enough.

Reply:
Fine by me. I'm chill with the duplication